src/index.ts

```diff
@@ -347,18 +348,35 @@ export class MayanRoute<N extends Network>
   }
 
   public override async *track(receipt: R, timeout?: number) {
-    if (!isSourceInitiated(receipt)) throw new Error("Transfer not initiated");
-
-    const txstatus = await getTransactionStatus(
-      receipt.originTxs[receipt.originTxs.length - 1]!
-    );
-    if (!txstatus) return;
+    const start = Date.now();
+
+    // What should be the default if no timeout is provided?
+    let leftover = timeout ? timeout : 60 * 60 * 1000;
+    while (leftover > 0) {
+      if (!isSourceInitiated(receipt))
+        throw new Error("Transfer not initiated");
+      try {
+        const txstatus = await getTransactionStatus(
+          receipt.originTxs[receipt.originTxs.length - 1]!
+        );
+        if (!txstatus) continue;
+
+        receipt = txStatusToReceipt(txstatus);
+        yield { ...receipt, txstatus };
+      } catch (e) {
+        // get transaction status checks for 404 and returns null
+        // if we get an error here its something else
+        throw e;
```

Comment:
whats the point of the try/catch block at all if you're just throwing it and doing nothing else

Reply:
removed